crates/driver/src/settlement_proposal.rs

```diff
@@ -123,6 +123,7 @@ pub struct SettlementProposal {
     pub clearing_prices: HashMap<H160, U256>,
     pub trades: Vec<TradedOrder>,
     pub execution_plan: Vec<Arc<dyn InteractionProposal>>,
+    pub id: u64,
```

Comment:
nit: Should we maybe call this `auction_id` so it's clear it's the same thing as `auction_id` in driver (I think you also used `auction_id` in SettlementSumary)?